### PR TITLE
add error test for batch write

### PR DESCRIPTION
**Pull request by marsishandsome**

### CODE REVIEW

File: core/src/main/scala/com/pingcap/tispark/TiBatchWrite.scala [hunk 1]
```diff
@@ -127,6 +127,15 @@ object TiBatchWrite {
     val kvClient = tiSession.createTxnClient()
     val tableRef = TiTableReference(options.database, options.table)
     val (tiDBInfo, tiTableInfo) = getDBAndTableInfo(tableRef, tiContext)
+
+    // check check check
```
Comment (birdstorm) [inline core/src/main/scala/com/pingcap/tispark/TiBatchWrite.scala]: LOL

File: core/src/test/scala/com/pingcap/tispark/datasource/ErrorSuite.scala [hunk 2]
```diff
@@ -0,0 +1,126 @@
+package com.pingcap.tispark.datasource
+
+import com.pingcap.tikv.exception.TiBatchWriteException
+import org.apache.spark.sql.Row
+import org.apache.spark.sql.types.{IntegerType, LongType, StringType, StructField, StructType}
+
+// without TiExtensions
+// will not load tidb_config.properties to SparkConf
+class ErrorSuite extends BaseDataSourceSuite("test_datasource_error") {
```
Comment (birdstorm) [inline core/src/test/scala/com/pingcap/tispark/datasource/ErrorSuite.scala]: should we rename it as `ExceptionTestSuite`?

Comment (marsishandsome) [reply core/src/test/scala/com/pingcap/tispark/datasource/ErrorSuite.scala]: nice, will rename in next patch

**Pull request merged**